tests/utils/host_utils.h

```diff
@@ -246,6 +271,24 @@ inline double getTolerance(QudaPrecision prec)
   return 1.0;
 }
 
+/**
+  @brief Return the negative exponent of the tolerances defined in `getTolerance` above
+  @param[in] prec Precision
+  @return Exponent of the base-10 reasonably expected tolerance
+*/
+inline int getNegLog10Tolerance(QudaPrecision prec)
```

Comment:
I think you should rewrite `getTolerance` to use `getNegLog10Tolerance` (or vice versa), to ensure that these are always consistent.

Reply:
Rewrote `getTolerance` as a function of `getNegLog10Tolerance` because `int` -> `double` seemed safer than the other way around. I checked the C++11+ standard and confirmed `int` will get promoted to `double` in this context. (I also confirmed it via a `ctest`.)

https://github.com/lattice/quda/pull/1367/commits/4722ed6db404269e99e7b9b6dc9d1b313bb4b4be